website/oauth/models/__init__.py

```diff
@@ -443,3 +443,66 @@ def absolute_api_v2_url(self):
     # used by django and DRF
     def get_absolute_url(self):
         return self.absolute_api_v2_url
+
+class ApiOAuth2PersonalToken(StoredObject):
+    """Information for user-created personal access tokens
+
+    This collection is also used by CAS to create the master list of available tokens.
+    Any changes made to field names in this model must be echoed in the CAS implementation.
+    """
+    _id = fields.StringField(primary=True,
+                             default=lambda: str(ObjectId()))
+
+    token_id = fields.StringField(default=functools.partial(random_string, length=70),
+                               unique=True)
+
+    owner = fields.ForeignField('User',
+                                backref='created',
+                                index=True,
+                                required=True)
+
+    name = fields.StringField(required=True, index=True)
+
+    scopes = fields.StringField(required=True)
```

Comment:
We ourselves got confused over whether this should be a list or a space-delimited string; a comment here would be helpful.